scripts/solve_network.py

```diff
@@ -189,44 +189,50 @@ def add_chp_constraints(n):
 def add_pipe_retrofit_constraint(n):
     """Add constraint for retrofitting existing CH4 pipelines to H2 pipelines."""
 
-    gas_pipes_i = n.links[n.links.carrier=="gas pipeline"].index
-    h2_retrofitted_i = n.links[n.links.carrier=='H2 pipeline retrofitted'].index
+    gas_pipes_i = n.links[n.links.carrier=="gas pipeline"].query("p_nom_extendable").index
+    h2_retrofitted_i = n.links[n.links.carrier=='H2 pipeline retrofitted'].query("p_nom_extendable").index
+    h2_retrofitted_fixed_i = n.links[n.links.carrier=='H2 pipeline retrofitted'].index.difference(h2_retrofitted_i)
```

Comment:
I think one can combine the query:
```suggestion
    gas_pipes_i = n.links.query("carrier == 'gas pipeline' and p_nom_extendable").index
    h2_retrofitted_i = n.links.query("carrier == 'H2 pipeline retrofitted' and p_nom_extendable").index
    h2_retrofitted_fixed_i = n.links.query("carrier == 'H2 pipeline retrofitted' and not p_nom_extendable").index
```